server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -313,7 +313,7 @@ private void beginSnapshot(final ClusterState clusterState,
                                final ActionListener<Snapshot> userCreateSnapshotListener) {
         threadPool.executor(ThreadPool.Names.SNAPSHOT).execute(new AbstractRunnable() {
 
-            boolean snapshotCreated;
+            final boolean snapshotCreated = false;
```

Comment:
Please double check this, and should we just remove it and pass `false` to the respective calls?

Reply:
Looks like it could be removed, see https://github.com/elastic/elasticsearch/commit/db7eb8e8fff57f03e5bcaa1e868463cfcaf1547b

But I'm also a bit confused since we removed the full block where it is set to true in https://github.com/crate/crate/commit/f5505cbfa113aa2a5612d910fde2e29bd919e06d

```
                if (clusterState.nodes().getMinNodeVersion().onOrAfter(NO_REPO_INITIALIZE_VERSION) == false) {
                    // In mixed version clusters we initialize the snapshot in the repository so that in case of a master failover to an
                    // older version master node snapshot finalization (that assumes initializeSnapshot was called) produces a valid
                    // snapshot.
                    repository.initializeSnapshot(
                        snapshot.snapshot().getSnapshotId(), snapshot.indices(), metaDataForSnapshot(snapshot, clusterState.metaData()));
                }
```

Maybe @michaelkleen can chime in.

Reply:
Since this block has been removed long ago, I think it's safe to remove the snapshotCreated var, and always make calls with `false`.

Reply:
WDYT?

Reply:
@mkleen @mfussenegger ?

Reply:
It looks this is only related to the error handling for `cleanupAfterError`. Now the metadata is always updated. I think this is safe remove as @matriv suggested.

Reply:
Thank you!